apps/spruce/src/pages/task/taskTabs/TaskHistory/index.tsx

```diff
@@ -1,12 +1,111 @@
+import { useState } from "react";
+import { useQuery } from "@apollo/client";
+import styled from "@emotion/styled";
+import {
+  SegmentedControl,
+  SegmentedControlOption,
+} from "@leafygreen-ui/segmented-control";
+import { H3 } from "@leafygreen-ui/typography";
+import { size } from "@evg-ui/lib/constants/tokens";
+import { useToastContext } from "@evg-ui/lib/context/toast";
+import {
+  TaskHistoryQuery,
+  TaskHistoryQueryVariables,
+  TaskQuery,
+} from "gql/generated/types";
+import { TASK_HISTORY } from "gql/queries";
+import { useQueryParam } from "hooks/useQueryParam";
+import { ACTIVATED_TASKS_LIMIT } from "./constants";
+import TaskTimeline from "./TaskTimeline";
+import { TaskHistoryOptions, ViewOptions } from "./types";
+import { groupTasks } from "./utils";
+
 interface TaskHistoryProps {
-  taskId: string;
-  execution: number;
+  task: NonNullable<TaskQuery["task"]>;
 }
 
-const TaskHistory: React.FC<TaskHistoryProps> = ({ execution, taskId }) => (
-  <div data-cy="task-history">
-    This is the history for task {taskId} and execution {execution}
-  </div>
-);
+const TaskHistory: React.FC<TaskHistoryProps> = ({ task }) => {
+  const dispatchToast = useToastContext();
+
+  const [viewOption, setViewOption] = useState(ViewOptions.Collapsed);
```

Comment:
This may be useful as a cookie value

Reply:
let's leave it without the cookie for now